specs/UnhandledKeyPressed.md

```diff
@@ -0,0 +1,234 @@
+
+# Examples
+## C++
+
+```cpp
+auto controller5 = m_controller.try_query<ICoreWebView2Controller5>();
+if (controller5)
+{
+    CHECK_FAILURE(controller5->add_UnhandledKeyPressed(
+        Callback<ICoreWebView2UnhandledKeyPressedEventHandler>(
+            [this](
+                ICoreWebView2Controller* sender,
+                ICoreWebView2UnhandledKeyPressedEventArgs* args) -> HRESULT
+            {
+                COREWEBVIEW2_KEY_EVENT_KIND kind;
+                CHECK_FAILURE(args->get_KeyEventKind(&kind));
+
+                // We only care about key down events.
+                if (kind == COREWEBVIEW2_KEY_EVENT_KIND_KEY_DOWN ||
+                    kind == COREWEBVIEW2_KEY_EVENT_KIND_SYSTEM_KEY_DOWN)
+                {
+                    COREWEBVIEW2_KEY_PRESSED_FLAG_KIND flag;
+                    CHECK_FAILURE(args->get_Modifiers(&flag));
+                    if (flag & COREWEBVIEW2_KEY_EVENT_FLAG_CONTROL_DOWN)
+                    {
+                        UINT key;
+                        CHECK_FAILURE(args->get_VirtualKey(&key));
+                        // Check if the key is one we want to handle.
+                        if (key == 'Z')
+                        {
+                            MessageBox(
+                                nullptr,
+                                L"Key combination Ctrl + Z unhandled by browser is "
+                                L"triggered.",
+                                L"", MB_OK);
+                        }
+                        else if (key >= 'A' && key <= 'Z')
+                        {
+                            OutputDebugString((std::wstring(L"Ctrl +") + (wchar_t)key +
+                                                L" not handled by browser is triggered.")
+                                                  .c_str());
+                        }
+                    }
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_unhandledKeyPressedToken));
+}
+```
+
+## C#
+
+```csharp
+private CoreWebView2Controller controller;
+void RegisterKeyEventHandlers()
+{
+    if (controller != null)
+    {
+        controller.UnhandledKeyPressed += CoreWebView2Controller_UnhandledKeyPressed;
+    }
+}
+void CoreWebView2Controller_UnhandledKeyPressed(object sender, CoreWebView2UnhandledKeyPressedEventArgs e)
+{
+    if ((e.KeyEventKind == CoreWebView2KeyEventKind.KeyDown) && (e.Modifiers & CoreWebView2KeyPressedFlagKind.CoreWebView2KeyEventFlagControlDown) != 0)
+    {
+        if (e.VirtualKey == 'z')
+        {
+            Debug.WriteLine($"Key combination Ctrl + Z unhandled by browser is triggered.");
+        }
+    }
+}
+```
+
+# API Details
+
+## Win32 C++
+```idl
+[uuid(053b9a5d-7033-4515-9898-912977d2fde8), object, pointer_default(unique)]
+interface ICoreWebView2Controller : IUnknown {
+  /// Adds an event handler for the `UnhandledKeyPressed` event.
+  /// `UnhandledKeyPressed` runs when an key is not handled in
+  /// the DOM.
```

Comment:
It would be good to explicitly say that this event will be raised after WebView2 has a chance to handle the key combination including the DOM.

Reply:
Modified the comment here to make it more clear here.